### DISCUSSION TITLE
`actions/runner` should provide Windows images for self-hosted Windows runners

### DISCUSSION THREAD

**Discussion by jrhemstad** [community/community#69324] [Actions]
### Select Topic Area

Product Feedback

### Body

As a user of GitHub Actions with self-hosted runners, I want to be able to self-host Windows runners.

For self-hosted Linux runners, I use the images provided by https://github.com/actions/runner, e.g., https://github.com/actions/runner/pkgs/container/actions-runner. 

While the `actions/runner` repo _does_ provide Windows binaries [in its releases](https://github.com/actions/runner/releases/tag/v2.309.0), it doesn't provide images that package those binaries [as is done for Linux](https://github.com/actions/runner/pkgs/container/actions-runner).

From looking at the [Dockerfile definition of the Linux images](https://github.com/actions/runner/blob/main/images/Dockerfile), it appears to be quite simple as it just downloads the relevant binaries from the corresponding release and untars them. 

I would like the `actions/runner` repo to provide the same thing for Windows images. 

From my research, it looks like there are a few people in the community who have already done this for themselves, e.g.,
- https://github.com/antgrutta/windows-runner-container
- https://github.com/isarkis/actions-runner-controller/pull/1/files

Related threads:
- https://github.com/actions/actions-runner-controller/issues/1001

Just to be clear, this is *not* a request for supporting Windows images in `jobs.<job_id>.container.image` as is [discussed here](https://github.com/actions/runner/issues/1402) (but I would like this too!). This is just for the `actions/runner` repo to build and provide Windows images as it does for Linux images.

**Comment by chenghuang-mdsol** [comment]
for old version runner: use this entrypoint script
```
WORKDIR "${RUNNER_WORKDIR}"
ENTRYPOINT [ "pwsh", "-c", "./config.cmd --name $env:RUNNER_NAME --url https://github.com/$env:RUNNER_REPO --token $env:RUNNER_TOKEN --labels $env:RUNNER_LABELS --unattended --replace --ephemeral; ./run.cmd"]
```

for scaletset: use this entrypoint
```
WORKDIR "${RUNNER_WORKDIR}"
& ./run.cmd --jitconfig $Env:ACTIONS_RUNNER_INPUT_JITCONFIG
```
